lib/puppet/application/lookup.rb

```diff
@@ -57,6 +57,102 @@ class Puppet::Application::Lookup < Puppet::Application
     end
   end
 
+  def help
+    <<-'HELP'
+
+puppet-lookup(8) -- Data in modules lookup function
+========
+
+SYNOPSIS
+--------
+The lookup command is used for debugging and testing a given data
+configuration. For a given data key, lookup will return which value
+will be produced with an optional explanation of how the system arrived
+at that value.
+
+USAGE
+-----
+puppet lookup [--help] [--type <TYPESTRING>] [--merge unique|hash|deep]
+  [--knock_out_prefix <PREFIX-STRING>] [--sort_merged_arrays]
+  [--unpack_arrays <STRING-VALUE>] [--merge_hash_arrays] [--explain]
+  [--default <VALUE>] [--node <NODE-NAME>] [--facts <FILE>] <keys>
+
+DESCRIPTION
+-----------
+The lookup command is a CLI interface for the puppet lookup function.
+When given one or more keys, the lookup command will return the first
+value found. The only component that is required in addition to the
+keys is a node (passed in via --node) which specifies which scope to
+perform the lookup in. It will exit with 0 if a value was returned
+and 1 otherwise.
+
```

Comment:
This is true when no explanation is wanted. The objective might well be to produce an explanation of why a value is not found so in PUP-4475, when explain is active, the exit code is always 0 (unless something is majorly wrong).